### DISCUSSION TITLE
customizing Info-default-directory-list

### DISCUSSION THREAD

**Discussion by martin-volf** [emacs-helm/helm#2573] [Q&A] [answered]
The change introduced with fixing #2118 causes that my customization of `Info-default-directory-list` does not work any longer. What's the expected way of doing that? The function `info-initialize` is called from `helm-info.el` early on, before my `init.el` file is read, and as a result, any changes done to `Info-default-directory-list` done in `init.el` are ignored.

**Comment by thierryvolpiatto** [comment, marked as answer]
Martin Volf ***@***.***> writes:

> The change introduced with fixing #2118 causes that my customization
> of Info-default-directory-list does not work any longer. What's the
> expected way of doing that? The function info-initialize is called
> from helm-info.el early on, before my init.el file is read,

How is this possible that helm-info is loaded before your init file?

> and as a result, any changes done to Info-default-directory-list done
> in init.el are ignored.
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.*Message ID: ***@***.***>


-- 
Thierry

**Reply by martin-volf** [reply]
> How is this possible that helm-info is loaded before your init file?

It is not, my bad. Something is calling `info-initialize` and the only reference I found is in helm-info, but that is indeed loaded long after the init file.

**Comment by thierryvolpiatto** [comment]
Martin Volf ***@***.***> writes:

>     How is this possible that helm-info is loaded before your init file?
>    
> It is not, my bad. Something is calling info-initialize and the only
> reference I found is in helm-info, but that is indeed loaded long
> after the init file.

info-initialize is not modifying Info-default-directory-list, only
Info-directory-list. If Info-directory-list is already feeded, it
doesn't touch it, otherwise it tries to give it a default value
according to INFOPATH if found and finally fallback to
(Info-default-dirs).
So I can't see for now in what Helm is involved here. Perhaps you can
give me more infos on how you are setting up your infos directories?

> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.*Message ID: ***@***.***>


-- 
Thierry

**Reply by martin-volf** [reply]
> info-initialize is not modifying Info-default-directory-list, only
Info-directory-list. If Info-directory-list is already feeded, it
doesn't touch it, otherwise it tries to give it a default value
according to INFOPATH if found and finally fallback to
(Info-default-dirs).

Right, so when something calls `info-initialize`, any further changes to `Info-default-directory-list` are ignored. I am adding custom directories to that default list in my init file and after a larger emacs and emacs-packages update this stopped to have any effect.

> So I can't see for now in what Helm is involved here.

Helm is not involved at all - I saw the reference to `info-initialize` in helm-info and concluded that this was the problem. But that conclusion was too quick; as you wrote, helm-info is loaded much later than my init file. Sorry for the confusion.

**Comment by thierryvolpiatto** [comment]
Martin Volf ***@***.***> writes:

>     info-initialize is not modifying Info-default-directory-list, only
>     Info-directory-list. If Info-directory-list is already feeded, it
>     doesn't touch it, otherwise it tries to give it a default value
>     according to INFOPATH if found and finally fallback to
>     (Info-default-dirs).
>    
> Right, so when something calls info-initialize, any further changes to
> Info-default-directory-list are ignored. I am adding custom
> directories to that default list in my init file and after a larger
> emacs and emacs-packages update this stopped to have any effect.

Info-default-directory-list and Info-directory-list are confusing, my
understanding is that Info-default-directory-list should not be a custom
variable, its purpose is to handle a list of defaults Emacs itself sets
on startup. Here I only modify Info-directory-list from my init file and
after that helm is loaded (in this case info-initialize is a noop).

>     So I can't see for now in what Helm is involved here.
>    
> Helm is not involved at all - I saw the reference to info-initialize
> in helm-info and concluded that this was the problem. But that
> conclusion was too quick; as you wrote, helm-info is loaded much later
> than my init file. Sorry for the confusion.

No problems.

> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.*Message ID: ***@***.***>


-- 
Thierry

**Comment by ghost** [comment]
Thierry Volpiatto ***@***.***> writes:

> Info-default-directory-list and Info-directory-list are confusing, my
> understanding is that Info-default-directory-list should not be a custom
> variable, its purpose is to handle a list of defaults Emacs itself sets
> on startup.

I think this warrants an M-x report-emacs-bug.  Martin, could you maybe
do that?

**Comment by thierryvolpiatto** [comment]
Michael Heerdegen ***@***.***> writes:

> Thierry Volpiatto ***@***.***> writes:
>
>> Info-default-directory-list and Info-directory-list are confusing, my
>> understanding is that Info-default-directory-list should not be a custom
>> variable, its purpose is to handle a list of defaults Emacs itself sets
>> on startup.
>
> I think this warrants an M-x report-emacs-bug. Martin, could you maybe
> do that?

I see that Info-directory-list is feeded in package.el, so if user set
Info-default-directory-list after package initialize (which is the way
to do) Info-default-directory-list will be ignored.

> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.*Message ID: ***@***.***>


-- 
Thierry

**Comment by martin-volf** [comment]
For the sake of completeness: looks like Info-default-directory-list should not be used the way I did it, and its comment says that, in a sense. I should have been using Info-additional-directory-list instead - that works without problems. The documentation in this respect is a bit hard to find (at least for the likes of me), but it is there.

**Comment by ghost** [comment]
Thierry Volpiatto ***@***.***> writes:

> I see that Info-directory-list is feeded in package.el, so if user set
> Info-default-directory-list after package initialize (which is the way
> to do) Info-default-directory-list will be ignored.

Do you also think that the user should just customize
`Info-additional-directory-list`?  Seems that this is indeed the option
one should normally change.
